File: dart/armstrong-numbers/lib/armstrong_numbers.dart
import 'dart:math';

class ArmstrongNumbers {
  bool isArmstrongNumber(num number) {
    final String chars = number.toString();

    final num sum =
        chars.split("").fold(0, (num previousValue, String element) {
      return previousValue + (pow(int.parse(element), chars.length));
    });

    return sum == number;
  }
}

File: java/armstrong-numbers/src/main/java/ArmstrongNumbers.java
class ArmstrongNumbers {

    boolean isArmstrongNumber(int numberToCheck) {

        final var str = Integer.toString(numberToCheck);
        int sum = 0;

        for (int i = 0; i < str.length(); i++) {
            final var ch = str.charAt(i);

            sum += (int) Math.pow(ch - '0', str.length());
        }

        return sum == numberToCheck;
    }

}
